app/models/grda_warehouse/ch_enrollment.rb

```diff
@@ -74,5 +74,324 @@ def self.update_existing!
         )
       end
     end
+
+    # Accept an optional date which will be used for extending the homeless
+    # range if the project is a homeless project
+    def self.chronically_homeless_at_start?(enrollment, date: enrollment.EntryDate)
+      chronically_homeless_at_start(enrollment, date: date) == :yes
+    end
+
+    # Was the client chronically homeless at the start of this enrollment?
+    # Optionally accepts a date to use for "CH at a point-in-time" calculation.
+    #
+    # @return [Symbol] :yes, :no, :dk_or_r, or :missing
+    def self.chronically_homeless_at_start(enrollment, date: enrollment.EntryDate)
+      chronically_homeless_at_start_steps(enrollment, date: date).last[:result]
+    end
+
+    # Was the client chronically homeless at the start of this enrollment?
+    # Optionally accepts a date to use for "CH at a point-in-time" calculation.
+    #
+    # @return [Array] all steps evaluated with result and display values.
+    #
+    # Each step in the array has this shape:
+    # {
+    #    result: Symbol, true, false, or nil
+    #    display_value: value to display in chronic-at-entry explanation table
+    #    line: [Number] line number in HUD calculation
+    # }
+    #
+    # Result "nil" means continue processing
+    # Result "false" means continue processing and skip branch
+    # Result "true" means continue processing and enter branch
+    # The last item in the array will have one of (:yes, :no, :dk_or_r, or :missing) as the result
+    def self.chronically_homeless_at_start_steps(enrollment, date: enrollment.EntryDate)
+      steps = []
+      # Line 1
+      steps.push(disabling_condition(enrollment))
+      return steps if steps.last[:result]
```

Comment:
If I understand this, it's going to return if `steps.last[:result]` is `true` or a symbol.  In this case it looks like `disabling_condition` returns a symbol for No, Doesn't know, or Missing, and a `nil` for Yes.  Which, I think would be correct.  Short circuit if the client doesn't have a disabling condition.  (Just checking my understanding.)

Reply:
That's correct, but I think it is a bit confusing because we never actually want to return early because of a `true`. I think the approach proposed in your/my other comment addresses this nicely.